File: Two Sum/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> res(2);
        unordered_map<int, int> map(numbers.size());
        for (int i = 0; i < numbers.size(); ++i) {
            auto it = map.find(target - numbers[i]);
            if (it != map.end()) {
                res[0] = it->second;
                res[1] = i + 1;
                return res;
            } else {
                map[numbers[i]] = i + 1;
            }
        }
        return res;
    }
};

File: Two Sum/Two Sum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] res = new int[2];
        Integer index = -1;
        for (int i = 0; i < numbers.length; ++i) {
            if ((index = map.get(target - numbers[i])) != null) {
                res[0] = index;
                res[1] = i + 1;
                return res;
            } else {
                map.put(numbers[i], i + 1);
            }
        }
        return res;
    }
}